### ISSUE TITLE
[CI/CD]: Review coverage of accessibility checks in discharge-wizard end-to-end tests

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#7347]
**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

Applications **must** have thorough end-to-end tests that run in our continuous integration/continuous deployment (CI/CD) pipeline. These tests should include thorough axe checks. While auditing the `/discharge-wizard` app folder, I noticed two places we could improve our test coverage by opening accordion menus, then running the axe check. I'd like the front-end engineering team to review this application, and update the tests to review this hidden content. Definition of done in acceptance criteria below.

## Point of Contact

**VFS Point of Contact:** _Jennifer_

## Environment
* `vets-website/src/applications/discharge-wizard/tests`
* `$ yarn test:e2e src/applications/discharge-wizard/tests`

## Acceptance Criteria

**Definition of done:**
- [ ] Front-end team member(s) have reviewed end-to-end tests for axe checks
- [ ] Open accordions, then run the `/discharge-upgrade-instructions/` axe check
- [ ] Open accordions, then run the `/discharge-upgrade-instructions/guidance/` axe check
- [ ] FE team has consulted with accessibility specialist in cases where there are high numbers of modals, accordions, other hidden content that could slow down e2e test runs.
- [ ] No axe `violations` appear in the plugin console. These will break the CI/CD build.

## WCAG or Vendor Guidance (optional)

* [Custom axeCheck helper method](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/testing/e2e/nightwatch-commands/axeCheck.js)
* [VSP guidance on writing end-to-end tests](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/platform/quality-assurance/e2e-testing)

**Comment by 1Copenut** [comment]
@luke-gcio If the team has moved to use the Cypress forms testing, would you mind commenting here and at include me? The Cypress forms tester has axe checks built in and would be a good way to close this issue without additional work.

**Comment by 1Copenut** [comment]
@luke-gcio Bumping for visibility. Thank you!

**Comment by 1Copenut** [comment]
@vhenry07 I'm making another pass through our end-to-end tests to verify if teams are using Cypress for current testing where possible, and have `cy.axeCheck()` calls in those tests. If your team has a few minutes, would you mind taking a look at this issue and follow up with me? Thank you.

**Comment by joshkimux** [comment]
@Mottie do you know if this is something we would own/be responsible for? (or is this out of date?)

**Comment by Mottie** [comment]
If an e2e test is missing an axe check, the developer will see an ESLint warning (not an error). Looking at the `discharge-wizard` specifically, [I do see included axe checks](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/discharge-wizard/tests/e2e/discharge-wizard.cypress.spec.js) even though the ESLint warning is showing up 😭

**Comment by joshkimux** [comment]
@Mottie noooo that's so weird. Do you know of anyone we can go to for this to be fixed? My first (dumb) guess is to message platform support, but I'm not confident that's the next best step (I can totally do this if it is).

**Comment by joshkimux** [comment]
Closing!
